scipy/stats/tests/test_stats.py

```diff
@@ -4200,6 +4212,105 @@ def test_kurtosistest_too_few_samples():
+         1., 1., 1., 1., 1., 2., 1., 1., 1., 1., 2., 1., 1., 2., 1., 1.,
+         2., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 2., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+         2., 1., 1., 1., 1., 1., 1., 1., 1., 1., 2., 1., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 3., 1., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 1.]
+
+    y = [1., 1., 1., 1., 1., 1., 1., 2., 1., 2., 1., 1., 1.,
+         1., 2., 1., 1., 1., 2., 1., 1., 1., 1., 1., 2., 1., 1., 3., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 2., 1., 2., 1., 1., 1., 1.,
+         1., 1., 2., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+         1., 1., 2., 1., 1., 1., 1., 1., 2., 2., 1., 1., 2., 1., 1., 2.,
+         1., 2., 1., 1., 1., 1., 2., 2., 1., 1., 1., 1., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 2., 1., 1., 1., 1., 1., 2., 2., 2., 1.,
+         1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
+         1., 2., 1., 1., 2., 1., 1., 1., 1., 2., 1., 1., 1., 1., 1., 1.,
+         1., 1., 1., 1., 1., 1., 2., 1., 1., 1., 2., 1., 1., 1., 1., 1.,
+         1.]
+
+    x = np.array(x)
+    y = np.array(y)
+
+    a = np.array([210.052110, 110.190630, 307.918612])
+    b = np.array([436.08811482466416, 416.37397329768191, 179.96975939463582,
+                  197.8118754228619, 34.038757281225756, 138.54220550921517,
+                  128.7769351470246, 265.92721427951852, 275.6617533155341,
+                  592.34083395416258, 448.73177590617018, 300.61495185038905,
+                  187.97508449019588])
+
+    alt_two = 'x and y are sampled from different populations'
+    alt_less = 'x is sampled from a population of smaller values than y'
+    alt_greater = 'x is sampled from a population of larger values than y'
+
+    # p-values and statistic compared against wilcox.test from R stats
+    def test_two_tailed_exact(self):
+        res = stats.stats.mann_whitney_u(self.a, self.b)
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16, 0.703571428571428625), decimal=12)
+        assert_string_equal(res.alternative, self.alt_two)
+
+    def test_less_exact(self):
+        res = stats.stats.mann_whitney_u(self.a, self.b, alternative='less')
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (23, 0.694642857142857117), decimal=12)
+        assert_string_equal(res.alternative, self.alt_less)
+
+    def test_greater_exact(self):
+        res = stats.stats.mann_whitney_u(self.a, self.b, alternative='greater')
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16, 0.351785714285714313), decimal=12)
+        assert_string_equal(res.alternative, self.alt_greater)
```

Comment:
Considering [this issue](https://github.com/scipy/scipy/pull/4933/files#r542015981) I was wondering where these values are coming from. 

I confirmed that R does give these p-values for these alternatives. Apparently, the R functions have a different notion of what the alternatives "less" and "greater" mean.  

Consider [this ](https://rdrr.io/cran/asht/man/wmwTest.html) implementation of the MWU test in R. With this input:
```
library(asht)
a = c(210.052110, 110.190630, 307.918612)
b = c(436.08811482466416, 416.37397329768191, 179.96975939463582, 
      197.8118754228619, 34.038757281225756, 138.54220550921517,
      128.7769351470246, 265.92721427951852, 275.6617533155341,
      592.34083395416258, 448.73177590617018, 300.61495185038905,
      187.97508449019588)
wmwTest(a, b, "greater", exact=TRUE)
```
we get the pvalue of `0.3518` shown above. 

Suppose we were to increase the entries in `a`, making the entries in `a` more likely to exceed the entries in `b`. Based on my intuition (from e.g. t-tests) I would have expected the p-value to _decrease_, thinking that this is testing whether `a` is in some sense "greater than" `b`, and the result should get more significant. But that's not what happens:

```
a = c(210.052110, 110.190630, 307.918612) + 200
wmwTest(a, b, "greater", exact=TRUE)
``` 
we end up with pvalue `0.9268`. Increase `a` further and the pvalue continues to go up.

Reply:
Actually it depends on which R function you use.  In the comments of the test, it states
```
p-values and statistic compared against wilcox.test from R stats
```

The convention is reversed there.

```
a = c(210.052110, 110.190630, 307.918612)
b = c(436.08811482466416, 416.37397329768191, 179.96975939463582, 
      197.8118754228619, 34.038757281225756, 138.54220550921517,
      128.7769351470246, 265.92721427951852, 275.6617533155341,
      592.34083395416258, 448.73177590617018, 300.61495185038905,
      187.97508449019588)

exact = TRUE

res1 = wilcox.test(a, b, alternative="greater", paired=FALSE, exact=exact )
res2 = wilcox.test(a, b, alternative="less", paired=FALSE, exact=exact )

print(res1$p.value)
print(res2$p.value)
```
prints
```
[1] 0.6946429
[1] 0.3517857
```